test/Microsoft.ML.CpuMath.UnitTests.netcoreapp/UnitTests.cs

```diff
@@ -75,433 +84,681 @@ public CpuMathUtilsUnitTests()
+            { "1", "-1.7", disableAvx, DisableAvxEnvironmentVariables },
+
+            { "0", "1.7", disableAvxAndSse, DisableAvxAndSseEnvironmentVariables },
+            { "1", "1.7", disableAvxAndSse, DisableAvxAndSseEnvironmentVariables },
+            { "0", "-1.7", disableAvxAndSse, DisableAvxAndSseEnvironmentVariables },
+            { "1", "-1.7", disableAvxAndSse, DisableAvxAndSseEnvironmentVariables },
+#endif
+        };
+
+        public static TheoryData<string, string, string, string, Dictionary<string, string>> MatMulData => new TheoryData<string, string, string, string, Dictionary<string, string>>()
+        {
+            {"0", "0", "0", defaultMode, null },
+            {"1", "1", "0", defaultMode, null },
+            {"1", "0", "1", defaultMode, null },
+#if NETCOREAPP3_0
+            {"0", "0", "0", disableAvx, DisableAvxEnvironmentVariables },
+            {"1", "1", "0", disableAvx, DisableAvxEnvironmentVariables },
+            {"1", "0", "1", disableAvx, DisableAvxEnvironmentVariables },
+
+            {"0", "0", "0", disableAvxAndSse, DisableAvxAndSseEnvironmentVariables },
+            {"1", "1", "0", disableAvxAndSse, DisableAvxAndSseEnvironmentVariables },
+            {"1", "0", "1", disableAvxAndSse, DisableAvxAndSseEnvironmentVariables },
+#endif
+        };
+
         [Theory]
-        [InlineData(0, 0, 0, new float[] { -416.6801f, -416.6801f, -416.6801f, -416.6801f, -416.6801f, -416.6801f, -416.6801f, -416.6801f })]
-        [InlineData(1, 1, 0, new float[] { 1496f, 3672f, 5848f, 8024f, 10200f, 12376f, 14552f, 16728f })]
-        [InlineData(1, 0, 1, new float[] { 204f, 492f, 780f, 1068f, 1356f, 1644f, 1932f, 2220f, 2508f, 2796f, 3084f, 3372f, 3660f, 3948f, 4236f, 4524f })]
-        public void MatMulTest(int matTest, int srcTest, int dstTest, float[] expected)
+        [MemberData(nameof(MatMulData))]
+        public void MatMulTest(string matTest, string srcTest, string dstTest, string mode, Dictionary<string, string> environmentVariables)
         {
-            AlignedArray mat = _testMatrices[matTest];
-            AlignedArray src = _testSrcVectors[srcTest];
-            AlignedArray dst = _testDstVectors[dstTest];
+            RemoteExecutor.RemoteInvoke((arg0, arg1, arg2, arg3) =>
+            {
+                CheckProperFlag(arg0);
+                AlignedArray mat = _testMatrices[int.Parse(arg1)];
+                AlignedArray src = _testSrcVectors[int.Parse(arg2)];
+                AlignedArray dst = _testDstVectors[int.Parse(arg3)];
 
-            CpuMathUtils.MatrixTimesSource(false, mat, src, dst, dst.Size);
-            float[] actual = new float[dst.Size];
-            dst.CopyTo(actual, 0, dst.Size);
-            Assert.Equal(expected, actual, _matMulComparer);
+                float[] expected = new float[dst.Size];
+                for (int i = 0; i < dst.Size; i++)
+                {
+                    float dotProduct = 0;
+                    for (int j = 0; j < src.Size; j++)
```

Comment:
You changed to calculating the expected results at runtime - because it's cleaner?